lib/network-context.tsx

```
@@ -0,0 +1,106 @@
+    const handleOffline = () => {
+      setNetworkState('offline');
+    };
+    
+    // Only attach event listeners on client side
+    if (typeof window !== 'undefined') {
+      // Register event listeners
+      window.addEventListener('online', handleOnline);
+      window.addEventListener('offline', handleOffline);
+      
+      // Performance monitoring
+      if ('connection' in navigator && (navigator as any).connection) {
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)